crates/sui-graphql-rpc/src/types/transaction_block.rs

```diff
@@ -438,27 +556,79 @@ impl TransactionBlockFilter {
             })?,
         })
     }
+
+    /// Returns count of filters that would require a `scan_limit` if used together.
+    pub(crate) fn num_filters(&self) -> usize {
+        [
+            self.recv_address.is_some(),
+            self.input_object.is_some(),
+            self.changed_object.is_some(),
+            self.function.is_some(),
+            self.kind.is_some(),
+        ]
+        .iter()
+        .filter(|&is_set| *is_set)
+        .count()
+    }
+
+    pub(crate) fn is_consistent(&self) -> Result<(), Error> {
```

Comment:
In all of these cases, we shouldn't return an error, we should be returning an empty result -- they are cases where we can statically determine that the result will be empty. The only new case where we should produce an error is when there are multiple filters and no scan limit.

Reply:
Might be nicer to return an error so the caller knows what's wrong and is fixable about their request, but it'd also make sense to just return an empty response